From 6103f6fc56f978c847ba7c1f2d9f38ee93a5e337 Mon Sep 17 00:00:00 2001
From: Sonja Heinze <sonjaleaheinze@gmail.com>
Date: Tue, 9 Mar 2021 12:57:47 +0100
Subject: [PATCH] Adapt to Ppxlib's API change

Ppxlib is removing Lexer.keyword_table from the API in exchange for
the more lightweight Keyword.is_keyword.

Signed-off-by: Sonja Heinze <sonjaleaheinze@gmail.com>
--- a/ppx_variants_conv.opam
+++ b/ppx_variants_conv.opam
@@ -15,7 +15,7 @@ depends: [
   "base"        {>= "v0.14" & < "v0.15"}
   "variantslib" {>= "v0.14" & < "v0.15"}
   "dune"        {>= "2.0.0"}
-  "ppxlib"      {>= "0.14.0"}
+  "ppxlib"      {>= "0.23.0"}
 ]
 synopsis: "Generation of accessor and iteration functions for ocaml variant types"
 description: "
--- a/src/ppx_variants_conv.ml
+++ b/src/ppx_variants_conv.ml
@@ -66,7 +66,7 @@ end
 
 let variant_name_to_string v =
   let s = String.lowercase v in
-  if Caml.Hashtbl.mem Lexer.keyword_table s
+  if Keyword.is_keyword s
   then s ^ "_"
   else s
 
